tests/PHPStan/Parser/RichParserTest.php

```diff
@@ -267,47 +267,171 @@ public function dataLinesToIgnoreParseErrors(): iterable
 			'<?php' . PHP_EOL .
 			'test(); // @phpstan-ignore return.ref,' . PHP_EOL,
 			[
-				2 => ['Unexpected trailing comma (,)'],
+				2 => ['Unexpected token type T_END after T_COMMA, expected T_IDENTIFIER'],
 			],
 		];
 
 		yield [
 			'<?php' . PHP_EOL .
 			'test(); // @phpstan-ignore test (comment),' . PHP_EOL,
 			[
-				2 => ['Unexpected trailing comma (,)'],
+				2 => ['Unexpected token type T_END after T_COMMA, expected T_IDENTIFIER'],
 			],
 		];
 
 		yield [
 			'<?php' . PHP_EOL .
 			'test(); // @phpstan-ignore ,' . PHP_EOL,
 			[
-				2 => ['First token is not an identifier'],
+				2 => ['Unexpected token type T_COMMA after start, expected T_IDENTIFIER'],
 			],
 		];
 
 		yield [
 			'<?php' . PHP_EOL .
 			'test(); // @phpstan-ignore return.ref, return.non )foo',
 			[
-				2 => ['Closing parenthesis ")" before opening parenthesis "("'],
+				2 => ['Unexpected token type T_CLOSE_PARENTHESIS after T_IDENTIFIER, expected T_COMMA or T_END or T_OPEN_PARENTHESIS'],
 			],
 		];
 
 		yield [
 			'<?php' . PHP_EOL .
 			'test(); // @phpstan-ignore return.ref, return.non (foo',
 			[
-				2 => ['Unclosed opening parenthesis "(" without closing parenthesis ")"'],
+				2 => ['Unexpected token type T_END, unclosed opening parenthesis'],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore ()' . PHP_EOL,
+			[
+				2 => ['Unexpected token type T_OPEN_PARENTHESIS after start, expected T_IDENTIFIER'],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore ' . PHP_EOL,
+			[
+				2 => ['Unexpected token type T_END after start, expected T_IDENTIFIER'],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore ,' . PHP_EOL,
+			[
+				2 => ['Unexpected token type T_COMMA after start, expected T_IDENTIFIER'],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore čumim' . PHP_EOL,
+			[
+				2 => ["Unexpected token type T_OTHER 'čumim' after start, expected T_IDENTIFIER"],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore test ((inner)' . PHP_EOL,
+			[
+				2 => ['Unexpected token type T_END, unclosed opening parenthesis'],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore return.ref return.non',
+			[
+				2 => ['Unexpected token type T_IDENTIFIER after T_IDENTIFIER, expected T_COMMA or T_END or T_OPEN_PARENTHESIS'],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore identifier (comment))' . PHP_EOL,
+			[
+				2 => ['Unexpected token type T_CLOSE_PARENTHESIS after T_CLOSE_PARENTHESIS, expected T_COMMA or T_END'],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore mečoun two' . PHP_EOL,
+			[
+				2 => ["Unexpected token type T_OTHER 'čoun' after T_IDENTIFIER, expected T_COMMA or T_END or T_OPEN_PARENTHESIS"],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore mečoun' . PHP_EOL,
```

Comment:
The identifier grammar should be the same as any other PHP identifier grammar, ie. allow `\x80-\xff` to be valid identifier character - https://www.php.net/manual/en/language.variables.basics.php#language.variables.basics.

Reply:
No, it's very simple ASCII for good reasons https://github.com/phpstan/phpstan-src/blob/16d5b010c61b5149b1b12970a70f3bde1dcc8358/src/Analyser/Error.php#L19